Core/BonusBuilder.cs

```diff
@@ -0,0 +1,107 @@
+﻿namespace Primordially.Core
+{
+    /// <summary>
+    ///     Used for multiple modifications of a single variable.
+    ///     All additions will be performed on the variable that was original chosen by <see cref="Character.ModifyVariable" />
+    /// </summary>
+    public readonly struct BonusBuilder<TKind> where TKind : IImmutableVariableCollection<TKind>
+    {
+        private readonly TKind _collection;
+        private readonly string _name;
+
+        private readonly CharacterVariable _workingVariable;
+
+        public BonusBuilder(TKind collection, string name)
+        {
```

Comment:
NIT: this should probably just call the other ctor, so if we make initialization code of some kind we don't have to duplicate it.